python/spider-py/src/spider_py/core/taskgraph.py

```diff
@@ -0,0 +1,56 @@
+"""TaskGraph module for Spider."""
+
+from spider_py.core.task import Task, TaskId
+
+
+class TaskGraph:
+    """Represents a task graph in Spider."""
+
+    def __init__(self) -> None:
+        """Initializes an empty task graph."""
+        self.tasks: dict[TaskId, Task] = {}
+        # Dependency list consists of a list of tuples of
+        #   - parent task id
+        #   - child task id
+        self.dependencies: list[tuple[TaskId, TaskId]] = []
+        self.input_tasks: set[TaskId] = set()
+        self.output_tasks: set[TaskId] = set()
+
+    def add_task(
+        self, task: Task, parents: list[TaskId] | None = None, children: list[TaskId] | None = None
+    ) -> None:
+        """
+        Adds a task to the graph.
+        :param task: The task to add.
+        :param parents: The parent ids of the task. Must be already in the task graph.
+        :param children: The children ids of the task. Must be already in the task graph.
+        """
+        self.tasks[task.task_id] = task
+        if parents:
```

Comment:
```suggestion
        if parents is not None:
```
We usually check it explicitly